lib/context/engine-context.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const INTERNAL = Symbol("internal");
+
+class EngineContext {
+    constructor(engine) {
+        this[INTERNAL] = {};
+
+        this[INTERNAL].engine = engine;
+    }
+
+    set directory(value) {
+        this[INTERNAL].directory = value;
+    }
+    
+    get directory() {
+        return this[INTERNAL].directory;
+    }
+}
```

Comment:
we may want to add a method or storage object like `currentPlan(s)` that will track the plans currently being run

Reply:
I didn't add it on purpose, because for now we don't need it. (I'd expect the need to appear in the future).